packages/test/test-end-to-end-tests/src/test/blobsisAttached.spec.ts

```diff
@@ -0,0 +1,201 @@
+			assert.strictEqual(
+				container.attachState,
+				AttachState.Detached,
+				"container should be detached",
+			);
+			assert.strictEqual(
+				detachedDataStore.handle.isAttached,
+				false,
+				"data store handle should be detached",
+			);
+			assert.strictEqual(dds.handle.isAttached, false, "dds handle should be detached");
+			assert.strictEqual(blobHandle.isAttached, false, "blob handle should be detached");
+		};
+
+		it("all detached", async function () {
+			checkForDetachedHandles(map);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after map is set in root directory", async function () {
+			detachedDataStore.root.set(mapId, map.handle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after directory is set in root directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			checkForDetachedHandles(directory);
+		});
+
+		it("after blob handle is set in map", async function () {
+			detachedDataStore.root.set("map", map.handle);
+			map.set("my blob", blobHandle);
+			checkForDetachedHandles(map);
+		});
+
+		it("after blob handle is set in directory", async function () {
+			detachedDataStore.root.set(directoryId, directory.handle);
+			directory.set("my blob", blobHandle);
+			checkForDetachedHandles(directory);
+		});
+
+		itExpects(
+			"after container is attached with map",
+			ContainerCloseUsageError,
+			async function () {
+				detachedDataStore.root.set("map", map.handle);
+				map.set("my blob", blobHandle);
+				const attachP = container.attach(
+					provider.driver.createCreateNewRequest(provider.documentId),
+				);
+				if (!driverSupportsBlobs(provider.driver)) {
+					return assert.rejects(attachP, (err) => err.message === usageErrorMessage);
+				}
+				await attachP;
+				detachedBlobStorage.blobs.clear();
+				assert.strictEqual(
+					container.attachState,
+					AttachState.Attached,
+					"container should be attached",
+				);
+				assert.strictEqual(detachedDataStore.handle.isAttached, true);
+				assert.strictEqual(map.handle.isAttached, true);
+				assert.strictEqual(blobHandle.isAttached, true);
+			},
+		);
+
+		itExpects(
+			"after container is attached with directory",
+			ContainerCloseUsageError,
+			async function () {
+				detachedDataStore.root.set(directoryId, directory.handle);
+				directory.set("my blob", blobHandle);
+				const attachP = container.attach(
+					provider.driver.createCreateNewRequest(provider.documentId),
+				);
+				if (!driverSupportsBlobs(provider.driver)) {
+					return assert.rejects(attachP, (err) => err.message === usageErrorMessage);
+				}
+				await attachP;
+				detachedBlobStorage.blobs.clear();
+				assert.strictEqual(container.attachState, AttachState.Attached);
+				assert.strictEqual(detachedDataStore.handle.isAttached, true);
+				assert.strictEqual(directory.handle.isAttached, true);
+				assert.strictEqual(blobHandle.isAttached, true);
+			},
+		);
+	});
```

Comment:
can you add a scenario:
- detached dds with blob and detached container, container attaches, then dds attaches

Reply:
the blob shouldn't attach until after the dds attaches